client/LICENSE

```diff
@@ -0,0 +1,4 @@
+The client module is dual licensed under MIT and Apache-2.0.
```

Comment:
No strong feelings on how we lay this out. Should we be using this though https://github.com/ipfs/ipfs-repository-template/blob/main/LICENSE.md?

Reply:
Good question. 

I copied this from what is widely used in JS ecosystem ([example: js-libp2p/auto-tls](https://github.com/libp2p/js-libp2p/tree/main/packages/auto-tls), it is the same for every other package there).  cc @achingbrain  for visibility in case 

My understanding is that automatons prefer licenses to be specified via explicit `SPDX-License-Identifier` hint, or  be places  in separate files following convention `LICENSE-FOO.md` introduced by [github in 2022](https://github.blog/changelog/2022-05-26-easily-discover-and-navigate-to-multiple-licenses-in-repositories/), but this detection only works if text file matches license, and one in JS land does not match reference APACHE by gthub:

>  ![image](https://github.com/user-attachments/assets/936d737c-4bdc-4768-ba81-fa865162a44f)

Don't want to sink any more time on this, I've switched to `SPDX-License-Identifier` way in this PR.

Reply:
I've posted some follow up [here](https://github.com/ipfs/ipfs-repository-template/issues/5) instead of doing on this closed PR.